[PM-17562] Add memory cache for event integrations (#6017)
bitwarden/server: src/Core/AdminConsole/Services/Implementations/EventIntegrations/README.md

    C1 -->|Can contain| C2[IntegrationFilterGroup...]
```

## Caching

To reduce database load and improve performance, integration configurations are cached in-memory during event
processing. Without caching, each incoming `EventMessage` would trigger a database query to retrieve the
relevant `OrganizationIntegrationConfigurationDetails`, followed by JSON deserialization of both the
configuration object and the optional filters. This repeated work becomes costly under high-throughput conditions.

The caching system is centered around the `IntegrationConfigurationDetailsCache`. It uses `IMemoryCache` to store

> configuration details keyed by the organization id, integration type and event type.

Reviewer: ```suggestion
configuration details keyed by the organization ID, integration type, and event type.
```

### What is cached

> Each cache entry is a deserialized list of `CachedIntegrationConfigurationDetails<T>` which include:

Reviewer: ```suggestion
Each cache entry is a deserialized list of `CachedIntegrationConfigurationDetails<T>` that includes:
```

> - The deserialized configuration (`T`).

Reviewer: ⛏ Drop all the "The"s.

- The deserialized IntegrationFilterGroup (if any).
- The integration template string.

If the relevant configuration set is not already cached, it is fetched using the
`IOrganizationIntegrationConfigurationRepository`, deserialized, validated, and stored in the cache. This avoids
both repeated deserialization of JSON strings and expensive round-trips to the database.

### Expiration strategy

The cache employs two expiration values to manage how long items stay in memory:

- **Absolute expiration** caps the maximum lifetime of a cache entry, regardless of usage.
    - Configurable via `GlobalSettings.EventLogging.CacheAbsoluteExpiration`.
    - Default: 60 minutes.
- **Sliding expiration** keeps active entries alive as long as they continue to be accessed.
    - Configurable via `GlobalSettings.EventLogging.CacheSlidingExpiration`.
    - Default: 10 minutes.

> There are also two methods for forcibly evicting entries from the cache (i.e. when configuration changes and

Reviewer: ```suggestion
There are also two methods for forcibly evicting entries from the cache (e.g. when configuration changes and
```
